lib/matplotlib/image.py

```diff
@@ -1374,26 +1383,28 @@ def imsave(fname, arr, vmin=None, vmax=None, cmap=None, format=None,
     Parameters
     ----------
     fname : str or file-like
-        Path string to a filename, or a Python file-like object.
-        If *format* is *None* and *fname* is a string, the output
-        format is deduced from the extension of the filename.
+        The filename or a Python file-like object to store the image in.
     arr : array-like
-        An MxN (luminance), MxNx3 (RGB) or MxNx4 (RGBA) array.
-    vmin, vmax: [ None | scalar ]
+        The image data. The shape can be one of
+        MxN (luminance), MxNx3 (RGB) or MxNx4 (RGBA).
+    vmin, vmax : scalar, optional
         *vmin* and *vmax* set the color scaling for the image by fixing the
         values that map to the colormap color limits. If either *vmin*
         or *vmax* is None, that limit is determined from the *arr*
         min/max value.
-    cmap : matplotlib.colors.Colormap, optional
-        For example, ``cm.viridis``.  If ``None``, defaults to the
-        ``image.cmap`` rcParam.
-    format : str
-        One of the file extensions supported by the active backend.  Most
-        backends support png, pdf, ps, eps and svg.
-    origin : [ 'upper' | 'lower' ]
-        Indicates whether the ``(0, 0)`` index of the array is in the
-        upper left or lower left corner of the axes.  Defaults to the
-        ``image.origin`` rcParam.
+    cmap : str or `~matplotlib.colors.Colormap`, optional
+        A Colormap instance or registered colormap name. The colormap
+        maps scalar data to colors. It is ignored for RGB(A) data.
+        Defaults to :rc:`image.cmap` ('viridis').
+    format : str, optional
+        The file format. One of the file extensions supported by the active
+        backend.  Most backends support png, pdf, ps, eps and svg.
```

Comment:
"One of the file extensions supported by the active backend.  Most backends support png, pdf, ps, eps and svg."
The first sentence is incorrect (and thus the second unnecessary): the backend will be automatically switched if needed.  For example Agg only supports png but you can save pdfs with Agg active, as a temporary pdf canvas will be created on the fly.
If you *really* want to be pedantic the correct list would be "formats either supported by the active backend, or listed in FigureCanvasBase.filetypes", but in practice it's just whatever savefig supports and I wouldn't bother repeating the list here.